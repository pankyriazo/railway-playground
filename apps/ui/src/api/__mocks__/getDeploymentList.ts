import { GetDeploymentListResponse } from "@/api/types";

export const mockGetDeploymentList = {
  data: {
    deployments: {
      edges: [
        {
          node: {
            id: "4b049de0-1ce0-4e81-8036-26b252e3285c",
            status: "SUCCESS",
            updatedAt: "2024-03-17T08:50:14.973Z",
            environmentId: "632da482-c5b1-4570-87af-9ea225c335cb",
          },
        },
        {
          node: {
            id: "38b68be4-f86d-4987-8c48-dee293fda404",
            status: "SUCCESS",
            updatedAt: "2024-03-16T19:41:25.747Z",
            environmentId: "1beb4373-6b1f-43cb-8150-0e7214acea82",
          },
        },
      ],
    },
  },
} as const satisfies GetDeploymentListResponse;

export const mockGetDeploymentListError = {
  data: null,
  errors: [
    {
      message: "Unauthorized",
    },
  ],
} as const satisfies GetDeploymentListResponse;
