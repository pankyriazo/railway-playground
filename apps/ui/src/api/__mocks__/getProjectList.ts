import { GetProjectListResponse } from "@/api/types";

export const mockGetProjectList = {
  data: {
    me: {
      projects: {
        edges: [
          {
            node: {
              id: "a20a0d44-a6de-4804-acf1-58d436570fd7",
              name: "nauseating-cup",
              services: {
                edges: [
                  {
                    node: {
                      id: "0e62e7bf-2008-4452-bd32-e57cec3bb9a2",
                      name: "fascinated-motion",
                    },
                  },
                  {
                    node: {
                      id: "7ab42902-aae8-47b7-9976-81496dc61834",
                      name: "Postgres",
                    },
                  },
                ],
              },
              plugins: {
                edges: [],
              },
              environments: {
                edges: [
                  {
                    node: {
                      id: "1beb4373-6b1f-43cb-8150-0e7214acea82",
                      name: "production",
                    },
                  },
                  {
                    node: {
                      id: "632da482-c5b1-4570-87af-9ea225c335cb",
                      name: "staging",
                    },
                  },
                ],
              },
            },
          },
        ],
      },
    },
  },
} as const satisfies GetProjectListResponse;

export const mockGetProjectListError = {
  data: null,
  errors: [
    {
      message: "Unauthorized",
    },
  ],
} as const satisfies GetProjectListResponse;
