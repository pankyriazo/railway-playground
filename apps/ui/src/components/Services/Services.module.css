.title {
  color: var(--color-text--600);
  font-size: var(--font-size-md);
  font-weight: var(--font-weight-medium);
  margin-bottom: var(--spacing-md);
}

.list {
  align-items: flex-start;
  display: flex;
  flex-direction: column;
  flex-wrap: nowrap;
  gap: var(--spacing-lg);
  overflow-x: auto;
  overflow-y: hidden;
  margin-top: var(--spacing-sm);
}

@media (min-width: 48rem) {
  .list {
    flex-direction: row;
  }
}
