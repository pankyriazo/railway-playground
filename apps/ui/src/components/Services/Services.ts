import { Edges, Node } from "@monorepo/types";
import { Component } from "@/types";
import Service from "@/components/Service";

import styles from "./Services.module.css";

class Services extends Component<HTMLUListElement> {
  private _project: Node;
  private _environment: Node;
  private _services: Edges<Node> = { edges: [] };

  constructor({
    project,
    services,
    environment,
  }: {
    project: Node;
    services: Edges<Node>;
    environment: Node;
  }) {
    super();

    this._element = document.createElement("ul");
    this._element.id = "services";
    this._project = project;
    this._environment = environment;
    this._services = services;

    this.render();
  }

  override render() {
    if (!this._element) {
      return;
    }

    this._element.innerHTML = "";
    this._element.dataset.testid = "services";

    const title = document.createElement("h4");
    title.className = styles.title!;
    title.textContent = "Services";

    this._element.appendChild(title);

    const list = document.createElement("ul");
    list.dataset.testid = "services__list";
    list.className = styles.list!;

    for (const { node: serviceData } of this._services.edges) {
      const service = new Service({
        project: this._project,
        service: serviceData,
        environment: this._environment,
      });
      list.appendChild(service.element);
    }

    this._element.appendChild(list);
  }
}

export default Services;
