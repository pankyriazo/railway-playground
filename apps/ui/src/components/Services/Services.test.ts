import { render } from "@/test-helpers";
import { mockGetProjectList } from "@/api/__mocks__/getProjectList";
import Services from "./Services";

describe("Services", () => {
  const project = mockGetProjectList.data.me.projects.edges[0].node;
  const mockProps = {
    project,
    services: project.services,
    environment: project.environments.edges[0].node,
  };

  beforeEach(() => {
    render(Services, mockProps);
  });

  it("should render", () => {
    const services = document.querySelector("[data-testid=services]");

    const serviceList = document.querySelector("[data-testid=services__list]");
    const serviceItems = serviceList?.querySelectorAll("li");

    expect(serviceItems).toHaveLength(2);
    expect(services).toMatchSnapshot();
  });
});
