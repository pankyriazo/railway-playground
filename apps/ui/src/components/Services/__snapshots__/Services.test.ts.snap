// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Services should render 1`] = `
<ul
  data-testid="services"
  id="services"
>
  <h4
    class="title"
  >
    Services
  </h4>
  <ul
    class="list"
    data-testid="services__list"
  >
    <li
      id="service-0e62e7bf-2008-4452-bd32-e57cec3bb9a2"
    />
    <li
      id="service-7ab42902-aae8-47b7-9976-81496dc61834"
    />
  </ul>
</ul>
`;
